test/integration/NarrativeTesting.js

```diff
@@ -0,0 +1,189 @@
+
+        return mergeObjects([caseData, envDefaults, envcaseData]);
+    }
+
+    /**
+    * Navigates the wdio browser to a workspace as given by its id, and waits until the 
+    * narrative container is visible.
+    * @arg {number} The narrative's workspace id
+    * @returns {Promise} The Promise value is ignored.
+    */
+    async openNarrative(workspaceId, timeoutOverride) {
+
+        // Go to the narrative!
+        const timeout = timeoutOverride || this.timeout;
+        await browser.url(makeURL(`narrative/${workspaceId}`));
+
+        // should experience loading blocker for a few seconds.
+        const loadingBlocker = await $('#kb-loading-blocker');
+        await loadingBlocker.waitForDisplayed({
+            timeout,
+            timeoutMsg: `Timeout after waiting ${timeout}ms for loading blocker to appear`
+        }); 
+
+        // And then the loading blocker should disappear!
+        await loadingBlocker.waitForDisplayed({
+            timeout,
+            timeoutMsg: `Timeout after waiting ${timeout}ms for loading blocker to disappear`,
+            reverse: true
+        });
+
+        // Ensure logged in
+        const loginButton = await $('#signin-button > div > button');
+        await loginButton.waitForDisplayed({
+            timeout,
+            timeoutMsg: `Timeout after waiting ${timeout}ms for login button to appear`
+        }); 
+
+        await clickWhenReady(loginButton);
+
+        const realnameElement = await $('[data-element="realname"]');
+        const usernameElement = await $('[data-element="username"]');
+        await browser.waitUntil(async () => {
+            const text = await realnameElement.getText();
+            return (text && text.length > 0);
+        });
+        const realname = await realnameElement.getText();
+        const username = await usernameElement.getText();
+        console.warn(`Signed in as user "${realname}" (${username})`);
+        await loginButton.click();
+    
+        // Ensure narrative notebook has displayed
+        // TODO: more interesting waitUntil loop to signal the 
+        // failure reason (useful for debugging tests?)
+        const container = await $('#notebook-container');
+        await container.waitForDisplayed({
+            timeout,
+            timeoutMsg: `Timeout after waiting ${timeout}ms for narrative to appear`
+        });
+        return container;
+    }
+
+    async waitForCell(notebookContainer, cellIndex){
+        return await browser.waitUntil(async () => {
+            const cell = await notebookContainer.$(`.cell:nth-child(${cellIndex})`);
+            return cell;
+        });
+    }
+
+    async selectCell(cell) {
+        // Make sure not selected.
+        // We do this by inspecting the border.
+        await browser.waitUntil(async () => {
+            const borderStyle = await cell.getCSSProperty('border');
+            return borderStyle.value === this.testData.common.unselectedBorder;
+        });
```

Comment:
Using the cell border to determine whether or not a cell is selected seems like a rather brittle way to test it -- if you are going to use cell styling to determine the state, I would have thought it would be better to see whether or not the cell has the class `selected`. You don't want presentation changes to affect the JS tests.